python/GafferImageUI/CatalogueUI.py

```diff
@@ -103,19 +105,90 @@ def registeredColumns() :
 
 	return __registeredColumns.keys()
 
+#
+# Value Providers
+#
+
+# Returns a column value provider that retrieves the value of a header from the
+# catalogue items image.  If multiple names are provided, the first present in
+# the image headers will be used, allowing a single column to support several
+# source headers depending on the image's origin.
+def imageHeaderValueProvider( nameOrNames, defaultValue = None ) :
+
+	if isinstance( nameOrNames, basestring ) :
+		nameOrNames = [ nameOrNames, ]
+
+	def provider( image, catalogue ) :
+		metadata = catalogue["out"].metadata()
+		for name in nameOrNames :
+			if name in metadata :
+				return metadata[ name ]
+
+		return defaultValue
+
+	return provider
+
+# Returns a column value provider that retrieves a render context value from
+# the catalogue items image.  If multiple names are provided, the first present
+# context entry will be used.
+def contextValueProvider( nameOrNames, defaultValue = None ) :
+
+	if isinstance( nameOrNames, basestring ) :
+		nameOrNames = [ nameOrNames, ]
+
+	headers = [ "gaffer:context:%s" % name for name in nameOrNames ]
+
+	return imageHeaderValueProvider( headers, defaultValue = defaultValue )
+
 #
 # Standard Columns
 #
 
-def __typeIconColumnValueProvider( image, catalogue ) :
-	return "catalogueTypeDisk" if image["fileName"].getValue() else "catalogueTypeDisplay"
+def __statusIconColumnValueProvider( image, catalogue ) :
+	return "catalogueStatusDisk" if image["fileName"].getValue() else "catalogueStatusDisplay"
 
-registerColumn( "typeIcon", "", __typeIconColumnValueProvider, ColumnType.Icon )
+registerColumn( "status", "", __statusIconColumnValueProvider, ColumnType.Icon )
 registerColumn( "name", "Name", lambda image, _ : image.getName() )
+registerColumn( "frame", "Frame", contextValueProvider( "frame" ) )
+registerColumn( "description", "Description", lambda image, _ : image["description"].getValue() )
+
+# Image properties
+
+def __resolutionColumnValueProvider( image, catalogue ) :
+	format_ = catalogue["out"].format()
+	value = "%d x %d" % ( format_.width(), format_.height() )
+	ratio = format_.getPixelAspect()
+	if ratio != 1.0 :
+		value += " [%d x %d]" % ( format_.width() * ratio, format_.height() )
```

Comment:
Agreed. Perhaps pixel aspect should just be a separate column?

Reply:
Oh, I see it is a separate column already. Let's just drop it from here then.

Reply:
It doesn't list pixel aspect here, it just shows the resulting resolution, which seemed like it might be useful. @andrewkaufman I tried generally speaking to match how we present it, but currently, we do things differently between 2D and 3D views anyway ( eg: `XxY` and `X x Y` ). We should unify those at some point.

Reply:
Removed in  https://github.com/GafferHQ/gaffer/pull/3688/commits/6c91dc3b6c9acfbb3dd02050157c9abb8f52992a#diff-f8b21d2701a672983fe238345811269bR195-R198